uxarray/grid.py

```diff
@@ -70,32 +68,88 @@ def __init__(self, dataset, **kwargs):
         for key in kwargs_list:
             setattr(self, key, kwargs.get(key, None))
 
-        # internal uxarray representation of mesh stored in internal object ds
-        self.ds = xr.Dataset()
+        # initialize face_area variable
+        self._face_areas = None
 
         # check if initializing from verts:
         if isinstance(dataset, (list, tuple, np.ndarray)):
             self.vertices = dataset
             self.__from_vert__()
             self.source_grid = "From vertices"
             self.source_datasets = None
-
         # check if initializing from string
         # TODO: re-add gridspec initialization when implemented
         elif isinstance(dataset, xr.Dataset):
+            self.ds = dataset
+            self.mesh_filetype = parse_grid_type(dataset, **kwargs)
```

Comment:
The `**kwargs` in the functon call was being input thorugh `open_dataset` before. I believe feeding the Grid constructor's `**kwargs` into `parse_grid_type ` wouldn't fit or be needed.